src/main/java/com/salesforce/storm/spout/dynamic/kafka/Consumer.java

```diff
@@ -442,98 +457,43 @@ private void fillBuffer() {
      *
      * @param outOfRangeException The exception that was raised by the consumer.
      */
-    private void handleOffsetOutOfRange(OffsetOutOfRangeException outOfRangeException) {
-        // Grab the partitions that had errors
-        final Set<TopicPartition> outOfRangePartitions = outOfRangeException.partitions();
+    private void handleOffsetOutOfRange(final OffsetOutOfRangeException outOfRangeException) {
         final Set<TopicPartition> resetPartitions = Sets.newHashSet();
 
-        // Grab all partitions our consumer is subscribed too.
-        Set<ConsumerPartition> allAssignedPartitions = getAssignedPartitions();
-
-        // Loop over all subscribed partitions
-        for (ConsumerPartition assignedConsumerPartition : allAssignedPartitions) {
-            // Convert to TopicPartition
-            final TopicPartition assignedTopicPartition = new TopicPartition(
-                assignedConsumerPartition.namespace(),
-                assignedConsumerPartition.partition()
-            );
-
-            // The last offset we went to start
-            final long lastStartedOffset = partitionOffsetsManager.getLastStartedOffset(assignedConsumerPartition);
-            final long lastFinishedOffset = partitionOffsetsManager.getLastFinishedOffset(assignedConsumerPartition);
-
-            // If this partition was out of range
-            // we simply log an error about data loss, and skip them for now.
-            if (outOfRangePartitions.contains(assignedTopicPartition)) {
-                // The last offset we have in our persistence layer
-                final long lastPersistedOffset = persistenceAdapter.retrieveConsumerState(
-                    getConsumerId(),
-                    assignedTopicPartition.partition()
-                );
-                // The offset that was in the error
-                final long exceptionOffset = outOfRangeException.offsetOutOfRangePartitions().get(assignedTopicPartition);
-
-                logger.error(
-                    "DATA LOSS ERROR - offset {} for partition {} was out of range, last started = {}, last persisted = {},"
-                    + " lastFinished = {}, original exception = {}",
+        // Loop over all the partitions in this exception
+        for (final TopicPartition topicPartition : outOfRangeException.offsetOutOfRangePartitions().keySet()) {
```

Comment:
Optional optimization here, You can do the following and avoid multiple calls into the map

```java
for (final Map.Entry<TopicPartition, Long>  entry : outOfRangeException.offsetOutOfRangePartitions().entrySet) {
    final TopicPartition topicPartition = entry.getKey();
    final long exceptionOffset = entry.getValue();
```